src/functions/module_lookup.ts

```diff
@@ -238,6 +245,76 @@ export const INDEX: AddFunctionDescription = {
   isExported: true,
 };
 
+// -----------------------------------------------------------------------------
+// INDEX
+// -----------------------------------------------------------------------------
+export const INDIRECT: AddFunctionDescription = {
+  description: _t("Returns the content of a cell, specified by row and column offset."),
+  args: [
+    arg("reference (string)", _t("The range of cells from which the values are returned.")),
+    arg(
+      "use_a1_notation (boolean, default=TRUE)",
+      _t(
+        "A boolean indicating whether to use A1 style notation (TRUE) or R1C1 style notation (FALSE)."
+      )
+    ),
+  ],
+  returns: ["ANY"],
+  compute: function (reference: FPayload, useA1Notation: Maybe<FPayload> = { value: true }): any {
+    let _reference = reference?.value?.toString();
+    if (!_reference) {
+      throw new InvalidReferenceError(_t("Reference should be defined."));
+    }
+    const sheetId = this.__originSheetId;
+    let originPosition: CellPosition | undefined;
+    const __originCellXC = this.__originCellXC?.();
+    if (__originCellXC) {
+      const cellZone = toZone(__originCellXC);
+      originPosition = {
+        sheetId,
+        col: cellZone.left,
+        row: cellZone.top,
+      };
+      // The following line is used to reset the dependencies of the cell, to avoid
+      // keeping dependencies from previous evaluation of the INDIRECT formula (i.e.
+      // in case the reference has been changed).
+      this.updateDependencies?.(originPosition);
+    }
+
+    const _useA1Notation = toBoolean(useA1Notation.value);
+    if (_useA1Notation === false) {
+      _reference = toA1(_reference, originPosition);
+    }
+    const range = this.getters.getRangeFromSheetXC(sheetId, _reference);
+    if (range === undefined || range.invalidXc || range.invalidSheetName) {
+      throw new InvalidReferenceError();
+    }
+    if (originPosition) {
+      this.addDependencies?.(originPosition, [range]);
+    }
+
+    const values: any[] = [];
+    for (let col = range.zone.left; col <= range.zone.right; col++) {
+      const colValues: any[] = [];
+      for (let row = range.zone.top; row <= range.zone.bottom; row++) {
+        const position = { sheetId: range.sheetId, col, row };
+        const value = this.getters.evaluateFormula(range.sheetId, toXC(col, row));
+        const evaluatedCell = this.getters.getEvaluatedCell(position);
+        if (evaluatedCell.type === CellValueType.error) {
+          throw evaluatedCell;
```

Comment:
use return as much as possible for errors, for performance reasons
```suggestion
          return evaluatedCell;
```

Reply:
here, we don't even need to check if we have an error or not, the formula must be able to spread errors